### PR TITLE
add e2e test for default label selector for NS in netpol

### PR DESCRIPTION
**Pull request by JornShen**
/kind feature

ref: https://github.com/kubernetes/kubernetes/issues/99552

```release-note
NONE
```

### CODE REVIEW

**Comment by mattfenwick** [comment]
pinging @jayunit100 and @knabben to get more 👀 on this !!!

**Comment by JornShen** [comment]
pull-kubernetes-e2e-kind failed two jobs:

Probing container should be ready immediately after startupProbe succeeds 
ref: https://github.com/kubernetes/kubernetes/issues/99979

Deployment should run the lifecycle of a Deployment
ref: https://github.com/kubernetes/kubernetes/issues/98180

pull-kubernetes-node-e2e

faile job: when querying /stats/summary should report resource usage through the stats api
ref: https://github.com/kubernetes/kubernetes/issues/96068


/test pull-kubernetes-node-e2e
/test pull-kubernetes-e2e-kind

**Comment by JornShen** [comment]
ping  @jayunit100 @knabben

**Comment by JornShen** [comment]
in jobs  pull-kubernetes-e2e-ubuntu-gce-network-policies:
these cases paased.
```
Kubernetes e2e suite: [sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client should enforce policy based on NamespaceSelector with MatchExpressions using immutable ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName] | 3m37s
Kubernetes e2e suite: [sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client should enforce policy to allow traffic based on NamespaceSelector with MatchLabels using immutable ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName]	1m40s
```

File: test/e2e/network/netpol/network_policy.go [hunk 1]
```diff
@@ -1088,6 +1088,47 @@ var _ = common.SIGDescribe("Netpol [LinuxOnly]", func() {
...
+
+			reachability := NewReachability(model.AllPods(), true)
+			reachability.ExpectPeer(&Peer{Namespace: nsX}, &Peer{Namespace: nsX, Pod: "a"}, false)
+			reachability.ExpectPeer(&Peer{Namespace: nsZ}, &Peer{Namespace: nsX, Pod: "a"}, false)
+			reachability.ExpectPeer(&Peer{Namespace: nsY}, &Peer{Namespace: nsX, Pod: "a"}, true)
+
+			ValidateOrFail(k8s, model, &TestCase{ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability})
+		})
+
+		ginkgo.It("should enforce policy based on NamespaceSelector with MatchExpressions using immutable ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName]", func() {
```
Comment (mattfenwick) [inline test/e2e/network/netpol/network_policy.go:1112]: Can you change `immutable` to `default` in this PR? 

The key to this test is that we're testing the default labels on namespaces.

**Review by mattfenwick — COMMENTED** [review]
Looks good, couple of minor changes and I'll be happy to approve!

File: test/e2e/network/netpol/network_policy.go [hunk 2]
```diff
@@ -1088,6 +1088,44 @@ var _ = common.SIGDescribe("Netpol [LinuxOnly]", func() {
 			reachability.ExpectAllIngress(NewPodString(nsX, "a"), false)
 			ValidateOrFail(k8s, model, &TestCase{ToPort: 81, Protocol: v1.ProtocolTCP, Reachability: reachability})
 		})
+
+		ginkgo.It("should enforce policy to allow traffic based on NamespaceSelector with MatchLabels using default ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName]", func() {
```
Comment (JornShen) [inline test/e2e/network/netpol/network_policy.go:1092]: @mattfenwick I am not sure that whether we need this label `[Feature:NamespaceDefaultLabelName]` ?

Comment (mattfenwick) [reply test/e2e/network/netpol/network_policy.go:1092]: yeah good question @rpkatz what do you think ?

Comment (mattfenwick) [reply test/e2e/network/netpol/network_policy.go:1092]: ah whoops pinged wrong handle ... @rikatz there we go

**Comment by mattfenwick** [comment]
/lgtm

Looks like the test failures were due to infra problems elsewhere!

**Comment by mattfenwick** [comment]
/approve

nice one @JornShen !

**Pull request merged**